### ISSUE TITLE
Regression: Mailparser 4.0.0 fails to parse an email that can be parsed in 3.15.0

### ISSUE THREAD

**Issue by seanthegeek** [SpamScope/mail-parser#123]
**Describe the bug**
Mailparser 4.0.0 fails to parse an email that can be parsed in 3.15.0

**To Reproduce**
1. `import mailparser`
2. `mail = mailparser.parse_from_string(s)`

**Expected behavior**
The email should be parsed

**Raw mail**
(https://github.com/domainaware/parsedmarc/blob/master/samples/forensic/%5BNetease%20DMARC%20Failure%20Report%5D%20Rent%20Reminder.eml)

**Environment**
 - OS: Linux
 - Docker: No
 - mail-parser version 4.0.0

**Additional context**

See https://github.com/domainaware/parsedmarc/actions/runs/11507562125/job/32033986931#step:8:223

**Comment by fedelemantuano** [comment]
The old version did see the following mail part:

```
--B_3622128044_155770608
Content-type: message/feedback-report; name="ATT00001";
 x-mac-creator="4F50494D"
Content-ID: <C9EED1E253B7F5459DB3AC82BC6E9953@namprd04.prod.outlook.com>
Content-disposition: attachment;
	filename="ATT00001"
Content-transfer-encoding: base64


RmVlZGJhY2stVHlwZTogYXV0aC1mYWlsdXJlDQpVc2VyLUFnZW50OiBOdGVzRG1hcmNSZXBv
cnRlci8xLjANClZlcnNpb246IDENCk9yaWdpbmFsLU1haWwtRnJvbTogPGJvdW5jZXMrMTEz
NzYxNi1jMWFkLXhzajM5OT0xNjMuY29tQGVtYWlsLmVudHJhdGEuY29tPg0KQXJyaXZhbC1E
YXRlOiBGcmksIDI4IFNlcCAyMDE4IDE2OjQ4OjQyICswODAwDQpTb3VyY2UtSVA6IDE2Ny44
OS42OS4yNA0KUmVwb3J0ZWQtRG9tYWluOiBjYXJkaW5hbC5jb20NCk9yaWdpbmFsLUVudmVs
b3BlLUlkOiBOOENvd0VBcGNVUG82cTFiblhsTUFBLS0uNDQzOTJTMw0KQXV0aGVudGljYXRp
b24tUmVzdWx0czogMTYzLmNvbTsgZGtpbT1wYXNzICh2ZXJpZnkgcmVzdWx0OiBhbGwgc2ln
bmF0dXJlcyB2ZXJpZmllZCkgaGVhZGVyLmQ9ZW50cmF0YS5jb207IHNwZj1wYXNzIHNtdHAu
bWFpbGZyb209Ym91bmNlcysxMTM3NjE2LWMxYWQteHNqMzk5PTE2My5jb21AZW1haWwuZW50
cmF0YS5jb20NCkRLSU0tRG9tYWluOiBlbnRyYXRhLmNvbQ0KRGVsaXZlcnktUmVzdWx0OiBk
ZWxpdmVyZWQNCklkZW50aXR5LUFsaWdubWVudDogc3BmLGRraW0=
--B_3622128044_155770608
Content-type: message/rfc822
Content-disposition: attachment

Received: by filter1356p1mdw1.sendgrid.net with SMTP id filter1356p1mdw1-30359-5BADEAE4-E
        2018-09-28 08:48:36.858842501 +0000 UTC m=+1856335.516585242
Received: from MTEzNzYxNg (unknown [198.190.14.10])
	by ismtpd0007p1las1.sendgrid.net (SG) with HTTP id oguVx1AxQayV5Sv2nYK-rA
	for <system@entrata.com>; Fri, 28 Sep 2018 08:48:36.716 +0000 (UTC)
From: 700 on Washington <info@cardinal.com>
To: "redacted@163.com" <redacted@163.com>
Subject: Rent Reminder
Thread-Topic: Rent Reminder
Thread-Index: AQHUVwgr2Hz5Jp/wMEGsfVZ94P1m5Q==
X-MS-Exchange-MessageSentRepresentingType: 1
Date: Fri, 28 Sep 2018 04:48:39 -0400
Message-ID: <oguVx1AxQayV5Sv2nYK-rA@ismtpd0007p1las1.sendgrid.net>
X-MS-Has-Attach:
X-MS-TNEF-Correlator:
X-MS-Exchange-Organization-RecordReviewCfmType: 0
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
--B_3622128044_155770608--
```

With the fix in #125, this part is parsed as attachment and the mail is parsed.

Try to use [this version](https://github.com/SpamScope/mail-parser/actions/runs/11604000335?pr=125).

I will wait your feedback.

**Comment by seanthegeek** [comment]
It is still failing

**Comment by fedelemantuano** [comment]
Just to clarify:

- I didn't push the change on Pypi
- The change is only on this branch
- You should test with [this binary](https://github.com/SpamScope/mail-parser/actions/runs/11604000335?pr=125)
- I run this command `curl -s https://raw.githubusercontent.com/domainaware/parsedmarc/refs/heads/master/samples/forensic/%5BNetease%20DMARC%20Failure%20Report%5D%20Rent%20Reminder.eml | mail-parser -k -j` and the change works

If you can confirm I need to see the error that you are having.

**Comment by seanthegeek** [comment]
I installed your patch using `pip install git+https://github.com/SpamScope/mail-parser.git@issue_123`

I am still getting the same error. Here's the traceback when I run `mailparser.parse_from_string()` for the sample. Not sure why the mailparser CLI acts differently.

```python
  File "/home/sean/dev/parsedmarc/parsedmarc/__init__.py", line 1180, in parse_report_email
    msg = mailparser.parse_from_string(input_)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sean/dev/parsedmarc/venv/lib/python3.11/site-packages/mailparser/mailparser.py", line 102, in parse_from_string
    return MailParser.from_string(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sean/dev/parsedmarc/venv/lib/python3.11/site-packages/mailparser/mailparser.py", line 219, in from_string
    return cls(message)
           ^^^^^^^^^^^^
  File "/home/sean/dev/parsedmarc/venv/lib/python3.11/site-packages/mailparser/mailparser.py", line 134, in __init__
    self.parse()
  File "/home/sean/dev/parsedmarc/venv/lib/python3.11/site-packages/mailparser/mailparser.py", line 468, in parse
    payload = payload.decode("raw-unicode-escape")
              ^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'decode'
```

**Comment by seanthegeek** [comment]
Never mind I had to uninstall the current version of mail-parser before it would actually install your patch branch. Your patch works.

Sorry about that,

**Comment by fedelemantuano** [comment]
Fixed with #125
